### PR TITLE
Make WordPressComServiceRemote use WordPressComRestApi

### PR DESCRIPTION
**Pull request by SergioEstevao**
Refs #5245 

To test:
 - Create a new wp.com account, by testing success scenarios and failure scenarios:
  - Repeated email
  - Invalid username
  - Invalid email
  - Insecure password
- Create a new wp.com site,  by testing success scenarios and failure scenarios:
 - Used site name
 - Site name is too short

Needs review: @koke

### CODE REVIEW

File: WordPress/Classes/Networking/WordPressComServiceRemote.m [hunk 1]
```diff
@@ -38,31 +38,12 @@ - (void)createWPComAccountWithEmail:(NSString *)email
     NSParameterAssert([username isKindOfClass:[NSString class]]);
     NSParameterAssert([password isKindOfClass:[NSString class]]);
     
-    void (^successBlock)(AFHTTPRequestOperation *, id) = ^(AFHTTPRequestOperation *operation, id responseObject) {
+    void (^successBlock)(id responseObject, NSHTTPURLResponse *httpResponse) = ^(id responseObject, NSHTTPURLResponse *httpResponse) {
```
Comment (koke) [inline WordPress/Classes/Networking/WordPressComServiceRemote.m]: You don't need to name parameters on the left side of the assignment as it's just a type annotation.

File: WordPress/Classes/Networking/WordPressComServiceRemote.m [hunk 2]
```diff
@@ -38,31 +38,12 @@ - (void)createWPComAccountWithEmail:(NSString *)email
...
-            NSString *localizedErrorMessage = [self errorMessageForError:error];
-            NSString *errorCode = [error.userInfo objectForKey:WordPressComApiErrorCodeKey];
-            NSMutableDictionary *userInfo = [[NSMutableDictionary alloc] initWithDictionary:error.userInfo];
-            [userInfo setValue:errorCode forKey:WordPressComApiErrorCodeKey];
-            [userInfo setValue:localizedErrorMessage forKey:WordPressComApiErrorMessageKey];
-            errorWithLocalizedMessage = [[NSError alloc] initWithDomain:error.domain code:error.code userInfo:userInfo];
-        }
-        
+    void (^failureBlock)(NSError *error, NSHTTPURLResponse *httpResponse) = ^(NSError *error, NSHTTPURLResponse *httpResponse){
+        NSError *errorWithLocalizedMessage = [self processError:error];
```
Comment (koke) [inline WordPress/Classes/Networking/WordPressComServiceRemote.m]: Maybe `processError:` should be `errorWithLocalizedMessage:`?

File: WordPress/WordPressApi/WordPressComRestApi.swift [hunk 3]
```diff
@@ -244,12 +249,19 @@ final class WordPressComRestAPIResponseSerializer: AFJSONResponseSerializer
     }
 
     override func responseObjectForResponse(response: NSURLResponse?, data: NSData?, error: NSErrorPointer) -> AnyObject? {
-        let responseObject = super.responseObjectForResponse(response, data: data, error: error)
+        var optionalError: NSError?
+        let responseObject = super.responseObjectForResponse(response, data: data, error: &optionalError)
+        error.memory = optionalError
```
Comment (koke) [inline WordPress/WordPressApi/WordPressComRestApi.swift]: From our chat yesterday, we talked about removing the custom `acceptableStatusCodes`. In that case, we shouldn't need to manually set the error, as the `super` implementation would already do it.

**Comment by koke** [comment]
Other than the comments, I tested this and seems to work fine

File: WordPress/WordPressApi/WordPressComRestApi.swift (hunk 3 above)
Comment (SergioEstevao) [inline WordPress/WordPressApi/WordPressComRestApi.swift]: @koke I'm doing this in order to have a variable with the error and grab the original userInfo from that error, before I add new things to it if needed.

**Comment by SergioEstevao** [comment]
@koke ready for another look.

File: WordPress/WordPressApi/WordPressComRestApi.swift (hunk 3 above)
Comment (koke) [inline WordPress/WordPressApi/WordPressComRestApi.swift]: You would still have that in `error.memory`, right? I read that differently the first time, and I think it's just two ways of writing the same thing. I still think we shouldn't set the `acceptableStatusCodes`.

You could leave here the original

```swift
let responseObject = super.responseObjectForResponse(response, data: data, error: error)
```

and further down just do

```swift
if let originalError = error.memory {
  userInfo = originalError.userInfo
}
```

**Comment by koke** [comment]
:shipit: if Travis passes

**Pull request merged**